src/core/readout.jl

```diff
@@ -0,0 +1,28 @@
+
+struct Readout <: AbstractInstruction
+    connected_qubit::Int
+end
+
+Base.inv(readout::Readout) = readout
+
+function get_connected_qubits(readout::Readout)::AbstractVector{Int}
+    return Vector{Int}([readout.connected_qubit])
+end
+
+function move_instruction(
+    readout::Readout,
+    qubit_mapping::AbstractDict{T,T},
+)::AbstractInstruction where {T<:Integer}
+
+    connected_qubits = get_connected_qubits(readout)
+
+    @assert length(connected_qubits) == 1 "a Readout can only be single-qubit"
+
+    qubit = connected_qubits[1]
+
+    if haskey(qubit_mapping, qubit)
+        qubit = qubit_mapping[qubit]
+    end
+
+    return Readout(qubit)
```

Comment:
I'm not sure I follow. The use case for `moved_instructions` is such as this one. If the `readout` stays in place there is not change to the result. This could be further optimized by moving `readouts` and other `Gates` and not doing the right-hand side `swaps`, however.
```
julia> c=QuantumCircuit(qubit_count=6, instructions=[control_x(1,6), Readout(6)])
Quantum Circuit Object:
   qubit_count: 6 
q[1]:──*───────
       |       
q[2]:──|───────
       |       
q[3]:──|───────
       |       
q[4]:──|───────
       |       
q[5]:──|───────
       |       
q[6]:──X────▽──
               



julia> transpile(transpiler,c)
Quantum Circuit Object:
   qubit_count: 6 
q[1]:──────────────────────*───────────────────────────
                           |                           
q[2]:─────────────────☒────X────☒──────────────────────
                      |         |                      
q[3]:────────────☒────☒─────────☒────☒─────────────────
                 |                   |                 
q[4]:───────☒────☒───────────────────☒────☒────────────
            |                             |            
q[5]:──☒────☒─────────────────────────────☒────☒───────
       |                                       |       
q[6]:──☒───────────────────────────────────────☒────▽──
```

Reply:
I am referring to `qubit_mapping`. We don't move the qubit unless the `qubit` is present in the `qubit_mapping`. I checked and the `move_instruction` for `Gate` also keeps a qubit as is, if a mapping isn't present. So I'm resolving this.